contracts/modules/assessment/Assessment.sol

```diff
@@ -2,515 +2,523 @@
+    uint cooldownPeriod,
+    uint groupId
+  ) external onlyContracts(C_GOVERNOR | C_COVER_PRODUCTS) {
+    uint length = productTypeIds.length;
+    for (uint i = 0; i < length; i++) {
+      _assessmentData[productTypeIds[i]] = AssessmentData({
+        assessingGroupId: groupId.toUint32(),
+        cooldownPeriod: cooldownPeriod.toUint32()
+      });
+    }
 
-    uint104 rewardsWithdrawableFromIndex = stakeOf[staker].rewardsWithdrawableFromIndex;
-    Vote memory vote;
-    Assessment memory assessment;
-    uint voteCount = votesOf[staker].length;
-    bool hasReachedUnwithdrawableReward = false;
-
-    for (uint i = rewardsWithdrawableFromIndex; i < voteCount; i++) {
-      vote = votesOf[staker][i];
-      assessment = assessments[vote.assessmentId];
-
-      // If hasReachedUnwithdrawableReward is true, skip and continue calculating the pending total
-      // rewards.
-      if (
-        !hasReachedUnwithdrawableReward &&
-        uint(assessment.poll.end) + PAYOUT_COOLDOWN >= block.timestamp
-      ) {
-        hasReachedUnwithdrawableReward = true;
-        // Store the index of the vote until which rewards can be withdrawn.
-        withdrawableUntilIndex = i;
-        // Then, also store the pending total value that can be withdrawn until this index.
-        withdrawableAmountInNXM = totalPendingAmountInNXM;
-      }
+    emit SetAssessmentDataForProductTypes(productTypeIds, cooldownPeriod, groupId);
+  }
+
+  /* ========== MUTATIVE FUNCTIONS ========== */
 
-      totalPendingAmountInNXM += uint(assessment.totalRewardInNXM) * uint(vote.stakedAmount) /
-        (uint(assessment.poll.accepted) + uint(assessment.poll.denied));
+  function addAssessorsToGroup(address[] calldata assessors, uint groupId) external onlyContracts(C_GOVERNOR) {
+    // make new group id
+    if (groupId == 0) {
+      groupId = ++_groupCount;
     }
 
-    if (!hasReachedUnwithdrawableReward) {
-      withdrawableUntilIndex = voteCount;
-      withdrawableAmountInNXM = totalPendingAmountInNXM;
+    uint length = assessors.length;
+    for (uint i = 0; i < length; i++) {
+      uint assessorMemberId = registry.getMemberId(assessors[i]);
+      require(assessorMemberId != 0, MustBeMember(assessors[i]));
+      _groups[groupId].add(assessorMemberId);
+      _groupsForAssessor[assessorMemberId].add(groupId);
+      emit AddAssessorToGroup(groupId, assessorMemberId);
     }
+  }
+
+  function setGroupMetadata(uint groupId, bytes32 ipfsMetadata) external onlyContracts(C_GOVERNOR) {
+    require(groupId > 0 && groupId <= _groupCount, InvalidGroupId());
 
-    return (totalPendingAmountInNXM, withdrawableAmountInNXM, withdrawableUntilIndex);
+    _groupsMetadata[groupId] = ipfsMetadata;
+    emit SetGroupMetadata(groupId, ipfsMetadata);
   }
 
-  /* === MUTATIVE FUNCTIONS ==== */
+  // todo: remove by address or by memberId?
+  function removeAssessorsFromGroup(address[] calldata assessors, uint groupId) external onlyContracts(C_GOVERNOR) {
```

Comment:
It's tempting to pass a single assessor instead of an array. The new Governor can have multiple actions at once, we can call this function multiple times.